frontend/src/bundles/common/components/slider/slider.tsx

```diff
@@ -0,0 +1,60 @@
+import { Box, Slider } from '@mui/material';
+import { styled } from '@mui/material/styles';
+import { useCallback } from 'react';
+
+import { getValidClassNames } from '../../helpers/helpers.js';
+import { defaultExperienceMarks } from './default-values.js';
+import classes from './style.module.css';
+import { defaultSliderStyle } from './styles.js';
+
+type Mark = {
+    label?: string;
+    value: number;
+};
+
+type Properties = {
+    label?: string;
+    value?: number;
+    marks?: Mark[] | boolean | undefined;
+    className?: string;
+};
+
+const CustomSlider: React.FC<Properties> = ({
+    label = 'slider label',
+    value,
+    marks = defaultExperienceMarks,
+    className = classes.sliderContainerStyle,
+}) => {
+    const StyledSlider = styled(Slider)(defaultSliderStyle);
+
+    const getValueLabel = useCallback(
+        (value: number): string | undefined => {
+            return Array.isArray(marks)
+                ? (marks.find((mark) => mark.value === value)?.label as string)
+                : 'undefined';
+        },
+        [marks],
+    );
+
+    return (
+        <Box
+            className={getValidClassNames(
+                className,
+                classes.sliderContainerStyle,
+            )}
+        >
+            <span>{label}</span>
+            <StyledSlider
+                aria-label={label}
+                defaultValue={value}
+                marks={marks}
+                step={null}
+                valueLabelDisplay="on"
```

Comment:
We don't need to pass _step, valueLabelDisplay_ through props ?

Reply:
Yes you are absolutely right. In our cases it correct props. But bare in mind, that we create general component which can be used not only for this situation. So you need to pass this variables through props, but make it unrequired and provide default values for this props which you need. That's why we can use this component whatever we want and provide new props for _step, valueLabelDisplay_ which we need in different situations. So main point of doing this is greater flexibility and reuse of the component

Reply:
ok, I will make changes